contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -193,19 +193,14 @@ contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterfac
       }
     }
 
-    bool success;
-    try
-      IReceiver(receiverAddress).onReport(
+    bool success = router.route(
+        combinedId,
+        msg.sender,
+        receiverAddress,
         rawReport[FORWARDER_METADATA_LENGTH:METADATA_LENGTH],
         rawReport[METADATA_LENGTH:]
```

Comment:
@DeividasK I could use some help, I'm running out of local variables (stack too deep). I've already rearranged whatever I could think of